collector/tcpstat.go

```diff
@@ -0,0 +1,210 @@
+// +build !notcpstat
+
+package collector
+
+import (
+	"bufio"
+	"fmt"
+	"io"
+	"os"
+	"strconv"
+	"strings"
+
+	"github.com/prometheus/client_golang/prometheus"
+)
+
+const (
+	procTcpStat  = "/proc/net/tcp"
```

Comment:
Sounds to me like reading `/proc` is the better option at least until it proves necessary to optimize further?